src/libstd/sys/windows/args.rs

```diff
@@ -12,32 +12,76 @@
 
 use os::windows::prelude::*;
 use sys::c;
-use slice;
-use ops::Range;
 use ffi::OsString;
-use libc::{c_int, c_void};
 use fmt;
+use collections::VecDeque;
+use core::iter;
 
 pub unsafe fn init(_argc: isize, _argv: *const *const u8) { }
 
 pub unsafe fn cleanup() { }
 
 pub fn args() -> Args {
     unsafe {
-        let mut nArgs: c_int = 0;
-        let lpCmdLine = c::GetCommandLineW();
-        let szArgList = c::CommandLineToArgvW(lpCmdLine, &mut nArgs);
-
-        // szArcList can be NULL if CommandLinToArgvW failed,
-        // but in that case nArgs is 0 so we won't actually
-        // try to read a null pointer
-        Args { cur: szArgList, range: 0..(nArgs as isize) }
+        let lp_cmd_line = c::GetCommandLineW();
+        let parsed_args_list = parse_lp_cmd_line(lp_cmd_line as *const u16);
+        Args { parsed_args_list: parsed_args_list }
     }
 }
 
+/// https://docs.microsoft.com/en-us/previous-versions//17w5ykft(v=vs.85)
```

Comment:
Could this perhaps have a bit more of a descriptive doc comment than simply a URL? For example what is this URL? What's the purpose of this function? Why aren't we using the `CommandLineToArgvW` function? etc